tests/integrations/object_detection/test_object_detection.py

```diff
@@ -0,0 +1,158 @@
+        scope="class",
+    )
+    def integration_manager(self, request):
+        manager = ObjectDetectionManager(config_path=request.param)
+        yield manager
+        manager.teardown()
+
+    @skip_inactive_stage
+    def test_train_complete(self, integration_manager):
+        # Test that file is created, model is loadable
+        manager = integration_manager
+        model_file = os.path.join(manager.save_dir.name, "exp", "weights", "last.pt")
+        assert os.path.isfile(model_file)
+        model, extras = load_checkpoint(
+            type_="val", weights=model_file, device=torch.device("cpu")
+        )
+        # Test that training ran to completion
+        assert extras["ckpt"]["epoch"] == -1
+
+    @skip_inactive_stage
+    def test_train_metrics(self, integration_manager):
+        # Test train metric(s) if specified
+        manager = integration_manager
+        train_args = manager.configs["train"]
+        model_file = os.path.join(manager.save_dir.name, "exp", "weights", "last.pt")
+        metrics, *_ = val(data=train_args.run_args.data, weights=model_file)
+        if "target_name" in train_args.test_args:
+            train_test_args = train_args.test_args
+            metric_idx = METRIC_TO_INDEX[train_test_args["target_name"]]
+            metric = metrics[metric_idx] * 100
+            target_mean = train_test_args["target_mean"]
+            target_std = train_test_args["target_std"]
+            assert target_mean - target_std <= metric <= target_mean + target_std
+
+    @skip_inactive_stage
+    def test_export_onnx_graph(self, integration_manager):
+        # Test that onnx model is loadable and passes onnx checker
+        manager = integration_manager
+        onnx_file = os.path.join(
+            os.path.dirname(manager.configs["export"]["args"].weights), "last.onnx"
+        )
+        assert os.path.isfile(onnx_file)
+        model = onnx.load(onnx_file)
+        onnx.checker.check_model(model)
+
+    @pytest.mark.skipif(not deepsparse, reason="Deepsparse not installed")
+    @skip_inactive_stage
+    def test_export_target_model(self, integration_manager):
+        # If target model provided in target_args, test that they have similar
+        # ort output
+        manager = integration_manager
+        export_args = manager.configs["export"]
+        target_model_path = export_args.test_args.get("target_model")
+        if not target_model_path:
+            pytest.skip("No target model provided")
+        run_model_path = os.path.join(
+            os.path.dirname(export_args.run_args.weights), "last.onnx"
+        )
+        _, *_ = load_checkpoint(
+            type_="val",
+            weights=target_model_path,
+            device=torch.device("cpu"),
+        )
+        input_data = deepsparse.utils.generate_random_inputs(run_model_path, 1)
+        input_names = deepsparse.utils.get_input_names(run_model_path)
+        output_names = deepsparse.utils.get_output_names(run_model_path)
```

Comment:
we shouldn't need to require deepsparse here. (see IC PR for this)

Reply:
Good call. Updated!